utils/restic/Makefile

```diff
@@ -0,0 +1,48 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=restic
+PKG_VERSION:=0.9.6
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/restic/restic/tar.gz/v${PKG_VERSION}?
+PKG_HASH:=1cc8655fa99f06e787871a9f8b5ceec283c856fa341a5b38824a0ca89420b0fe
+
+PKG_LICENSE:=BSD-2-Clause
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Markus Weippert <markus@gekmihesg.de>
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=github.com/restic/restic/
+GO_PKG_BUILD_PKG:=github.com/restic/restic/cmd/restic/
+
```

Comment:
Not sure if the version number is embedded in the binary. If it isn't, I think you can add this:

```
GO_PKG_LDFLAGS_X:=main.version=$(PKG_VERSION)
```

You can also add

```
GO_PKG_LDFLAGS:=-s -w
```

to mirror the upstream build process, though OpenWrt binaries are usually stripped anyway which should accomplish the same thing as adding these flags.

Reply:
main.version seems to be set here (from the VERSION file): https://github.com/restic/restic/blob/604b18aa7426148a55f76307ca729e829ff6b61d/build.go#L551
Flags are now set explicitly.

Reply:
The golang-package.mk build process doesn't use build.go. It calls `go install` directly.

build.go is used when [manually building from source](https://restic.readthedocs.io/en/latest/020_installation.html#from-source):

```
$ go run build.go
```

Reply:
I suggest running `restic version` to verify the version number is embedded correctly.

Reply:
Sorry, you're right of course. It was still working because it's explicitly set here: https://github.com/restic/restic/blob/b723ca3de5b5d7716993aad6dc8e04e9d0ea69fa/cmd/restic/global.go#L42
It isn't set for rest-server though, so I've added the line in both packages for consistency.